src/keas/kmi/testclient.py

```diff
@@ -15,33 +16,39 @@
 """
 __docformat__ = "reStructuredText"
 
+import os
 import sys
 import optparse
 import textwrap
-import urlparse
+try:
+    # Python 3
+    from urllib.parse import urlparse
+except ImportError:
+    # Python 2
+    from urlparse import urlparse
 
 from keas.kmi.facility import LocalKeyManagementFacility
 
 def ping(kmf):
-    pieces = urlparse.urlparse(kmf.url)
+    pieces = urlparse(kmf.url)
     conn = kmf.httpConnFactory(pieces.netloc)
     conn.request('GET', '/')
     response = conn.getresponse()
-    print response.status, response.reason
-    print
-    print response.read()
+    print(response.status, response.reason)
+    print()
+    print(response.read())
 
 
 def new_key(kmf):
-    sys.stdout.write(kmf.generate())
+    os.write(sys.stdout.fileno(), kmf.generate())
```

Comment:
I did try sys.stdout.buffer first, but when I saw sys.stdout.buffer didn't work on Python 2, I searched for another solution and found this.